packages/nimble-components/src/select/index.ts

```diff
@@ -39,61 +89,776 @@ export class Select extends FoundationSelect implements ErrorPattern {
+
+    /**
+     * Reference to the internal listbox element.
+     *
+     * @internal
+     */
+    public listbox!: HTMLDivElement;
+
+    /**
+     * The unique id for the internal listbox element.
+     *
+     * @internal
+     */
+    public listboxId: string = uniqueId('listbox-');
+
     /**
      * @internal
      */
     @observable
-    public region?: AnchoredRegion;
+    public scrollableElement!: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public filterInputElement?: HTMLInputElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public region!: AnchoredRegion;
 
     /** @internal */
     @observable
     public hasOverflow = false;
 
-    // Workaround for https://github.com/microsoft/fast/issues/5123
-    public override setPositioning(): void {
+    /**
+     * @internal
+     */
+    @observable
+    public filteredOptions: ListboxOption[] = [];
+
+    /**
+     * @internal
+     */
+    @observable
+    public filter = '';
+
+    /**
+     * @internal
+     */
+    @observable
+    public scrollbarIsVisible = false;
+
+    /**
+     * @internal
+     */
+    @observable
+    public committedSelectedOption: ListboxOption | undefined = undefined;
+
+    /**
+     * The max height for the listbox when opened.
+     *
+     * @internal
+     */
+    @observable
+    public maxHeight = 0;
+
+    /**
+     * The component is collapsible when in single-selection mode with no size attribute.
+     *
+     * @internal
+     */
+    @volatile
+    public get collapsible(): boolean {
+        return !(this.multiple || typeof this.size === 'number');
+    }
+
+    private _value = '';
+    private forcedPosition = false;
+    private indexWhenOpened?: number;
+
+    // This intersection observer is to handle focus behavior for when the dropdown becomes
+    // visible.
+    private readonly regionElementIntersectionObserver: IntersectionObserver = new IntersectionObserver(
+        entries => {
+            if (
+                entries.length > 0
+                    && entries[entries.length - 1]!.intersectionRatio > 0
+            ) {
+                this.focusAndScrollOptionIntoView();
+            }
+        },
+        { threshold: 1.0, root: document }
+    );
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.addEventListener('change', this.changeValueHandler);
+        this.addEventListener('contentchange', this.updateDisplayValue);
+        this.forcedPosition = !!this.positionAttribute;
+        this.regionElementIntersectionObserver.observe(this.region);
+        this.initializeOpenState();
+    }
+
+    public override disconnectedCallback(): void {
+        this.removeEventListener('change', this.changeValueHandler);
+        this.removeEventListener('contentchange', this.updateDisplayValue);
+        this.regionElementIntersectionObserver.unobserve(this.region);
+        super.disconnectedCallback();
```

Comment:
Is the ordering significant? If so can youu add a comment, if not can we call super first?